roles/common/tasks/rhel-entitlements.yml

```diff
@@ -14,26 +14,54 @@
   changed_when: false
   no_log: true
 
+- name: Set rhsm_registered
+  set_fact:
+    rhsm_registered: "{{ subscription.rc == 0 }}"
+  when: subscription is defined
+
 - name: Register with subscription-manager.
   command: subscription-manager register
            --activationkey={{ subscription_manager_activationkey }}
            --org={{ subscription_manager_org }}
   no_log: true
-  when: subscription is defined and
-        subscription.rc != 0
+  when: rhsm_registered == "False"
```

Comment:
I think I was confused because the set_fact above has a ``when: subscription is defined`` which led me to believe that rhsm_registered might not be created in some circumstances.